src/human_encoding/README.md

```diff
@@ -0,0 +1,189 @@
+# Simplicity Human-Readable Encoding
+
+This module defines a human-readable encoding for Simplicity programs. This encoding
+is intended to be the encoding used for storage and interchange of "commitment-time"
+Simplicity programs, i.e. programs which are unpruned and have no witnesses.
+
+The following parts of the encoding are incomplete/undesigned:
+
+1. It does not support witness data; in future we would like to support partial/full
+   witness population, as well as population of disconnected expressions.
+2. `2^n` when it appears in types is a single lexer token and you cannot put spaces
+   into it. I'm not sure if I want to fix this or not.
+
+With that said, the rest of the document defines the encoding.
+
+## Syntax
+
+The syntax is defined in `src/human_encoding/parse/ast.rs`. It currently uses the
+`santiago` parser generator, but we would like to move away from this, probably to
+an ad-hoc parser, to avoid poor asymptotic behavior and to get better error messages.
+
+Comments are started by `--` and end at the next newline. This is the only aspect
+in which whitespace is significant.
+
+Simplicity expressions are a series of **definitions** of the form
```

Comment:
aed42d0045f0403f05e263c737bf6dc3c4153d9a: The word "expression" has a different meaning here than in the rest of the document. A (semantic) Simplicity expression is defined by a series of (syntactic) definitions.

Maybe "A Simplicity file consists of a series of DEFINITIONS of the form..."

Reply:
Yes, "Simplicity expression" has a different meaning than "expression". I am careful about this but I agree that it's confusing and potentially misleading.